sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/data/cosmos/CosmosReactiveRepositoriesAutoConfiguration.java

```diff
@@ -19,7 +21,8 @@
 @ConditionalOnClass({ ReactiveCosmosRepository.class })
 @ConditionalOnMissingBean({ ReactiveCosmosRepositoryFactoryBean.class,
     ReactiveCosmosRepositoryConfigurationExtension.class })
-@ConditionalOnProperty(prefix = "azure.cosmos.repositories",
+@ConditionalOnBean(ReactiveCosmosTemplate.class)
```

Comment:
This auto-configuration must be put after the auto-configuration of ReactiveCosmosTemplate, using @ConfigurationAfter, to make sure the condition won't be evaluated before the auto-configuration of ReactiveCosmosTemplate